src/platform/apollolake/include/platform/memory.h

```diff
@@ -136,22 +136,22 @@
 #define L2_SRAM_BASE		0xA000A000
 #define L2_SRAM_SIZE		0x00056000
 
+#define MAX_BLOCK_SIZE		256
```

Comment:
Can this name be less generic ? To indicate it's related to HEAP, or even just for non-rballoc

Reply:
@RanderWang This will be platform and heap config specific. Use the buffer heap block size here.
```
HEAP_BUFFER_BLOCK_SIZE
```
You should also remodel the heap configs and remove any block sizes above HEAP_BUFFER_BLOCK_SIZE e.g.
```
/* Heap section sizes for module pool */
#define HEAP_RT_COUNT8		0
#define HEAP_RT_COUNT16		48
#define HEAP_RT_COUNT32		48
#define HEAP_RT_COUNT64		32
#define HEAP_RT_COUNT128	32
#define HEAP_RT_COUNT256	32
#define HEAP_RT_COUNT512	4 << remove since > than buffer block size.
#define HEAP_RT_COUNT1024	4 << remove
```
Removing the large block frees up 6k which can either be added to this heap as smaller blocks or automatically to the buffer heap.  I would add a 48, 96, 192 block sizes with the extra 6k as this makes the memory more granular.  
```
/* Heap section sizes for system runtime heap */
#define HEAP_SYS_RT_COUNT64	64
#define HEAP_SYS_RT_COUNT512	8 << remove
#define HEAP_SYS_RT_COUNT1024	4 << remove
```
This saves 8k, which again can be reallocated where best needed.

Reply:
@lgirdwood   Now the algorithm is:
````
   if (size >= 384)
          rballoc
   else
          _malloc
````
So HEAP_RT_COUNT384 should be added to make HEAP_RT  to support allocation size < 384 && size > 256

Reply:
@lgirdwood , I find that cache line size is 64byte in xtensa core, I would drop 48 and 96